### ISSUE TITLE
The solution file mbedTLS.sln is missing

### ISSUE THREAD

**Issue by nurbles** [Mbed-TLS/mbedtls#9081]
### Summary
The solution file mbedTLS.sln, (referenced in the readme.md file) is missing.  There is no solution or project to build this package with Visual Studio.

**Comment by tgonzalezorlandoarm** [comment]
Hello!
As described in the same README.md file,
"In the development branch of Mbed TLS, the Visual Studio solution files need to be generated first as described in [“Generated source files in the development branch”](https://github.com/Mbed-TLS/mbedtls/blob/development/README.md#generated-source-files-in-the-development-branch)."

**Comment by tgonzalezorlandoarm** [comment]
In future, this kind of question (Support requests) is better suited to the mailing list ( mbed-tls@lists.trustedfirmware.org )

**Comment by nurbles** [comment]
I have perl, Visual Studio and python.  I don't know what "some python packages" are, but I hope they were installed by the script.  However, the `make_generated_files.bat` script fails:
```
D:\code\mbedtls-development>python -m pip install --user -r scripts/basic.requirements.txt
Ignoring Jinja2: markers 'python_version < "3.10"' don't match your environment
Requirement already satisfied: markupsafe<2.1 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from -r scripts\driver.requirements.txt (line 7)) (2.0.1)
Requirement already satisfied: Jinja2>=2.10.3 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from -r scripts\driver.requirements.txt (line 15)) (3.1.3)
Requirement already satisfied: types-Jinja2>=2.11.9 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from -r scripts\driver.requirements.txt (line 17)) (2.11.9)
Requirement already satisfied: jsonschema>=3.2.0 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from -r scripts\driver.requirements.txt (line 18)) (4.22.0)
Requirement already satisfied: types-jsonschema>=3.2.0 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from -r scripts\driver.requirements.txt (line 19)) (4.22.0.20240501)
Requirement already satisfied: types-MarkupSafe in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from types-Jinja2>=2.11.9->-r scripts\driver.requirements.txt (line 17)) (1.1.10)
Requirement already satisfied: attrs>=22.2.0 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from jsonschema>=3.2.0->-r scripts\driver.requirements.txt (line 18)) (23.2.0)
Requirement already satisfied: jsonschema-specifications>=2023.03.6 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from jsonschema>=3.2.0->-r scripts\driver.requirements.txt (line 18)) (2023.12.1)
Requirement already satisfied: referencing>=0.28.4 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from jsonschema>=3.2.0->-r scripts\driver.requirements.txt (line 18)) (0.35.1)
Requirement already satisfied: rpds-py>=0.7.1 in c:\users\nurbl\appdata\roaming\python\python312\site-packages (from jsonschema>=3.2.0->-r scripts\driver.requirements.txt (line 18)) (0.18.0)

D:\code\mbedtls-development>scripts\make_generated_files.bat

D:\code\mbedtls-development>python scripts\generate_driver_wrappers.py   || exit /b 1

D:\code\mbedtls-development>perl scripts\generate_errors.pl   || exit /b 1

D:\code\mbedtls-development>perl scripts\generate_query_config.pl   || exit /b 1

D:\code\mbedtls-development>perl scripts\generate_features.pl   || exit /b 1

D:\code\mbedtls-development>python scripts\generate_ssl_debug_helpers.py   || exit /b 1

D:\code\mbedtls-development>perl scripts\generate_visualc_files.pl   || exit /b 1

D:\code\mbedtls-development>python scripts\generate_psa_constants.py   || exit /b 1

D:\code\mbedtls-development>python tests\scripts\generate_bignum_tests.py   || exit /b 1

D:\code\mbedtls-development>python tests\scripts\generate_ecp_tests.py   || exit /b 1

D:\code\mbedtls-development>python tests\scripts\generate_psa_tests.py   || exit /b 1
Traceback (most recent call last):
  File "D:\code\mbedtls-development\tests\scripts\generate_psa_tests.py", line 850, in <module>
    test_data_generation.main(sys.argv[1:], __doc__, PSATestGenerator)
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\test_data_generation.py", line 224, in main
    generator.generate_target(target)
  File "D:\code\mbedtls-development\tests\scripts\generate_psa_tests.py", line 846, in generate_target
    super().generate_target(name, self.info)
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\test_data_generation.py", line 177, in generate_target
    self.write_test_data_file(name, test_cases)
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\test_data_generation.py", line 164, in write_test_data_file
    test_case.write_data_file(filename, test_cases)
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\test_case.py", line 88, in write_data_file
    for tc in test_cases:
  File "D:\code\mbedtls-development\tests\scripts\generate_psa_tests.py", line 695, in all_test_cases
    if key.location_value() != 0:
       ^^^^^^^^^^^^^^^^^^^^
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\psa_storage.py", line 172, in location_value
    return self.lifetime.value() >> 8
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\psa_storage.py", line 75, in value
    self.update_cache()
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\psa_storage.py", line 47, in update_cache
    values = c_build_helper.get_c_expression_values(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\c_build_helper.py", line 152, in get_c_expression_values
    compile_c_file(c_name, exe_name, include_path)
  File "D:\code\mbedtls-development\tests\scripts\..\..\scripts\mbedtls_dev\c_build_helper.py", line 95, in compile_c_file
    proc = subprocess.Popen(cmd,
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Python312\Lib\subprocess.py", line 1538, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 2] The system cannot find the file specified

D:\code\mbedtls-development>
```
I _believe_ I followed the instructions, but they aren't completely clear to me... I _think_ I only needed to do the bullet step starting with "On Windows" but maybe there is more I needed to do?  Or some required python or perl packages that aren't actually mentioned or named?  Or this fails for some other reason?

**Comment by gilles-peskine-arm** [comment]
I recommend using a release instead of the `development` branch. Releases come with all platform-and-configuration-independent files generated.

I recognize the failure as a place where the Python script calls a C compiler. On Windows, you'll probably need to set the `CC` environment variable to point to a compiler. MSVC is supported, but its location is not automatically detected.

Most people on the Mbed TLS development team are used to Unix-like development environments and Unix-like or embedded targets, not so much to Windows. So often we don't know the “proper” way to do things on Windows. For example I wrote the code that throws this exception, and it works on my machine, but for Windows I only did the minimum to pass our CI. We welcome patches if you know your platform better.
